### PR TITLE
Refactor of settings.py

### PR DESCRIPTION
**Pull request by Brachi**
Changed several parts of the code to reduce verbosity and make it a little more readable. No functionality changed. Added 2 TODOs to discuss functions that return a string or None, to either update the docstrings or the functions + the tests

### CODE REVIEW

File: src/onelogin/saml2/settings.py [hunk 1]
```diff
@@ -557,18 +507,14 @@ def get_sp_cert(self):
...
-            if exists(cert_file_name):
-                f_cert = open(cert_file_name, 'r')
-                cert = f_cert.read()
-                f_cert.close()
+        if not cert and exists(cert_file_name):
+            with open(cert_file_name) as f:
+                cert = f.read()
 
-        return cert
+        return cert or None  # TODO: Should change rtype docstrings or update tests
```
Comment (Brachi) [inline src/onelogin/saml2/settings.py]: Can the docstring be updated? Not sure if changing the return type to be only string is a good idea at this stage.

**Comment by joac** [comment]
Code looks more pythonic, great job!

File: src/onelogin/saml2/settings.py (hunk 1 above)
Comment (pitbulk) [inline src/onelogin/saml2/settings.py]: I don't think is a good idea, let's keep return types (and only change that if strictly necessary)

**Comment by pitbulk** [comment]
I implemented that toolkit from the [php-saml](https://github.com/onelogin/php-saml) toolkit and I willfully tried to keep the structure between the toolkits (same with the java and dotnet) in order to help me in the maintenance. But I think this specific change that you proposed does not affect me to port future changes on the rest of the toolkits.

I will review and merge.

Thanks for contribute!  (I will apply those changes also on [python3-saml](https://github.com/onelogin/python3-saml)

File: src/onelogin/saml2/settings.py (hunk 1 above)
Comment (Brachi) [inline src/onelogin/saml2/settings.py]: I updated the docstrings and removed the TODO comments.

**Comment by Brachi** [comment]
@pitbulk you've done a terrific job maintaining this library in many languages, this PR was just to make it more readable to those used to Python conventions, and I agree the structure should be kept as much as possible (e.g. although in Python getters and setters are not common practice, I'd leave them there).
I will open an issue for discussion first for any changes more than just style.

**Comment by pitbulk** [comment]
Thanks @brachi , I read fast your comment and saw  the"terrific" word that alarmed me. This is a spanish ["false friend"](http://www.ziggurat.es/es/lecciones_ingles/index.asp?id=1116).

Thanks for contribute! Sure we can find a balance between portability and python code style :)

**Comment by pitbulk** [comment]
I tried to apply this change to python3-saml and the tests failed. I will need to review if we changed the behavior: https://github.com/onelogin/python3-saml/pull/24/files

**Comment by Brachi** [comment]
@pitbulk it seems tests/src/OneLogin/saml2_tests/response_test.py is failing at setup time in *both* master and the settings_refactor branches. If that test file is removed all other tests pass (including settings_test). Is that the failure you had?

**Comment by pitbulk** [comment]
The trace of the errors on python3-saml
```
======================================================================
ERROR: testGetSPMetadataSigned (tests.src.OneLogin.saml2_tests.settings_test.OneLogin_Saml2_Settings_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pitbulk/proyectos/python3-saml/tests/src/OneLogin/saml2_tests/settings_test.py", line 412, in testGetSPMetadataSigned
    self.generateAndCheckMetadata(settings_info)
  File "/home/pitbulk/proyectos/python3-saml/tests/src/OneLogin/saml2_tests/settings_test.py", line 431, in generateAndCheckMetadata
    metadata = compat.to_string(settings.get_sp_metadata())
  File "/home/pitbulk/proyectos/python3-saml/src/onelogin/saml2/settings.py", line 596, in get_sp_metadata
    OneLogin_Saml2_Error.PUBLIC_CERT_FILE_NOT_FOUND
OneLogin_Saml2_Error: Cannot sign metadata: missing SP public key certificate.

======================================================================
FAIL: testIsInValidLogoutResponseSign (tests.src.OneLogin.saml2_tests.auth_test.OneLogin_Saml2_Auth_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pitbulk/proyectos/python3-saml/tests/src/OneLogin/saml2_tests/auth_test.py", line 989, in testIsInValidLogoutResponseSign
    self.assertIn('In order to validate the sign on the SAMLResponse, the x509cert of the IdP is required', auth.get_errors())
AssertionError: 'In order to validate the sign on the SAMLResponse, the x509cert of the IdP is required' not found in ['invalid_logout_response_signature', 'Signature validation failed. Logout Response rejected']

======================================================================
FAIL: testIsValidLogoutRequestSign (tests.src.OneLogin.saml2_tests.auth_test.OneLogin_Saml2_Auth_Test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pitbulk/proyectos/python3-saml/tests/src/OneLogin/saml2_tests/auth_test.py", line 1077, in testIsValidLogoutRequestSign
    self.assertIn('In order to validate the sign on the SAMLRequest, the x509cert of the IdP is required', auth.get_errors())
AssertionError: 'In order to validate the sign on the SAMLRequest, the x509cert of the IdP is required' not found in ['invalid_logout_request_signature', 'Signature validation failed. Logout Request rejected']
```

**Comment by pitbulk** [comment]
never mind, the problem was in the tests :)

**Pull request merged**